### PR TITLE
Remove some unused headers and functions

### PR DESCRIPTION
**Pull request by huoyaoyuan**

### CODE REVIEW

File: src/coreclr/pal/prebuilt/inc/fxver.h [hunk 1]
```diff
@@ -1,7 +1,9 @@
 // Licensed to the .NET Foundation under one or more agreements.
 // The .NET Foundation licenses this file to you under the MIT license.
 
+#ifndef HOST_UNIX
 #include <verrsrc.h>
+#endif
```
Comment (huoyaoyuan) [inline src/coreclr/pal/prebuilt/inc/fxver.h:6]: The custom .rc parser on Unix seems honoring #include rule, but does not looks into the actual content of macros. Do we need a comment for it?

**Comment by huoyaoyuan** [comment]
Going to push more today

**Comment by huoyaoyuan** [comment]
This one is ready for review now. @janvorli any interest?

File: src/coreclr/pal/prebuilt/inc/fxver.h (hunk 1 above)
Comment (jkotas) [reply src/coreclr/pal/prebuilt/inc/fxver.h:6]: If this ifdef is tricky, a comment would be useful.

**Review by jkotas — APPROVED** [review]
LGTM. Thank you!

File: src/coreclr/pal/prebuilt/inc/fxver.h (hunk 1 above)
Comment (janvorli) [reply src/coreclr/pal/prebuilt/inc/fxver.h:8]: I am not sure I understand what you mean by "doesn't inspect the actual values of the macros". The parser first runs the resource file through the C preprocessor, so it should process all macros should be expanded to their values during that phase.

**Review by AaronRobinsonMSFT — APPROVED** [review]
Thank you!

File: src/coreclr/pal/prebuilt/inc/fxver.h (hunk 1 above)
Comment (huoyaoyuan) [reply src/coreclr/pal/prebuilt/inc/fxver.h:8]: I didn't look depth about how the parser works and how it works, but it seems that palrt.h doesn't bring any macro used. I'll update the comment.

**Comment by huoyaoyuan** [comment]
Is this fine now?

**Pull request merged**